File: cashmanager/src/com/example/model/Article.java
package com.example.model;

/**
 * jlegay 2019
 * #EPITECH - Article model server side {JAVA}
 */
public class Article {
    private String id;
    private String nom;
    private String description;
    private String img;
    private Float prix;
    public Article(String id,String nom,String description,String img, Float prix) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.img = img;
        this.prix = prix;
    }

    public String stringify() {
        return this.id+"#"+this.nom+"#"+this.description+"#"+this.img+"#"+this.prix.toString();
    }
    static Article parse(String stringify) {
        String[] astr = stringify.split("#");
        return new Article(
                astr[0],
                astr[1],
                astr[2],
                astr[3],
                Float.parseFloat(astr[4])
        );
    }
}

File: cashapp/app/src/main/java/com/example/cashapp/model/Article.kt
package com.example.cashapp.model

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.util.Log
import android.widget.ImageView
import io.grpc.internal.ReadableBuffers.openStream
import java.net.URL


class Article (
    val id: String,
    val nom: String,
    val description: String,
    val img: String,
    val prix: Float) {

    companion object {
        fun stringify(article : Article) : String {
            return "${article.id}#${article.nom}#${article.description}#${article.img}#${article.prix.toString()}"
        }

        fun parse(stringified : String) : Article {
            val astr = stringified.split('#')
            return Article(
                astr[0],
                astr[1],
                astr[2],
                astr[3],
                astr[4].toFloat()
            )
        }
    }

    inner class DownloadImageTask(internal var bmImage: ImageView) :
        AsyncTask<String, Void, Bitmap>() {

        override fun doInBackground(vararg urls: String): Bitmap? {
            if (urls.equals("noimg")) return null
            val urldisplay = urls[0]
            var mIcon11: Bitmap? = null
            try {
                val `in` = java.net.URL(urldisplay).openStream()
                mIcon11 = BitmapFactory.decodeStream(`in`)
            } catch (e: Exception) {
                Log.e("Error", e.message)
                e.printStackTrace()
            }

            return mIcon11
        }

        override fun onPostExecute(result: Bitmap?) {
            if (result is Bitmap)
                bmImage.setImageBitmap(result)
        }
    }
}